### PR TITLE
EdDSA support and tests

### LINKED ISSUE #196: Add support for EdDSA certificates
As EdDSA is part of TLS 1.3 standard and part of the backported changes to TLS 1.2, we should include test cases and test support for it.

Support both Ed25519 and Ed448.

See [RFC 8422](https://tools.ietf.org/html/rfc8422)

### LINKED ISSUE #253: TLS 1.3 Certificate Verify signatures (EdDSA) test script
verify that server is willing to sign the Certificate Verify message with a complete set of EdDSA signatures supported in TLS 1.3.

blocked by #196 

see also: #211, #252

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Add support for connecting to servers that use EdDSA certificates and for creating EdDSA signatures.

### Motivation and Context

fixes #196 
fixes #253 
depends on https://github.com/tlsfuzzer/tlslite-ng/pull/472

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [x] NSS - n/a, unsupported
  - [ ] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

### CODE REVIEW

File: scripts/test-tls13-eddsa-in-certificate-verify.py [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+
+        if supported is None:
+            return sigalg
+        if sigalg in supported:
+            return sigalg
+
+    raise ValueError(
+        "Couldn't find a supported Signature Algorithm that  matches the" +
+        " provided parameters: {0}, {1}, {3}".format(alg_type, hash_pref,
+                                                    cert_type))
```
Comment (inikolcev) [inline scripts/test-tls13-eddsa-in-certificate-verify.py:117]: I don't see this function used in the script.

File: scripts/test-eddsa-in-certificate-verify.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
+# Author: Hubert Kario, (c) 2021
+# Released under Gnu GPL v2.0, see LICENSE file for details
+"""Test for EdDSA support in Certificate Verify"""
```
Comment (inikolcev) [inline scripts/test-eddsa-in-certificate-verify.py:3]: Don't we also need to test here that fuzzed or empty signatures in CertificateVerify are correctly rejected with decrypt error?

File: tlsfuzzer/helpers.py [hunk 3]
```diff
@@ -71,8 +71,15 @@
 Subset of :py:const:`ECDSA_SIG_ALL`.
 """
 
+EDDSA_SIG_ALL = [SignatureScheme.ed448,
```
[the remark this answers is not in the document]
Comment (tomato42) [reply tlsfuzzer/helpers.py:74]: Not really, it only matters if the server has both Ed448 and Ed25519 key, and that's rather unlikely.
But for consistency with the other `*_ALL`, the strongest parameters are listed first.

File: tlsfuzzer/messages.py [hunk 4]
```diff
@@ -1046,6 +1046,12 @@ def _sig_alg_for_ecdsa_key(accept_sig_algs, version, key):
         # this is better last resort than sending a sha1+rsa sigalg
         return (getattr(HashAlgorithm, hash_name), SignatureAlgorithm.ecdsa)
 
+    @staticmethod
+    def _sig_alg_for_eddsa_key(key_alg, accept_sig_algs, version):
```
[the remark this answers is not in the document]
Comment (tomato42) [reply tlsfuzzer/messages.py:1050]: true

File: scripts/test-eddsa-in-certificate-verify.py [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+    print("                usage: [-x probe-name] [-X exception], order is compulsory!")
+    print(" -n num         run 'num' or all(if 0) tests instead of default(10)")
+    print("                (excluding \"sanity\" tests)")
+    print(" -k file.pem    file with private key for client")
+    print(" -c file.pem    file with certificate for client")
+    print(" --help         this message")
+
+
+def main():
+    """check if obsolete signature algorithm is rejected by server"""
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-eddsa-in-certificate-verify.py:62]: yes, it's a copy-paste leftover; fixed

File: scripts/test-tls13-eddsa-in-certificate-verify.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-tls13-eddsa-in-certificate-verify.py:117]: yes, leftover from copy-paste, fixed

**Review by inikolcev — APPROVED** [review]
